### DISCUSSION TITLE
Why does Actor.onTouchEvent{} ignore touchDragged events?

### DISCUSSION THREAD

**Discussion by DLochmelis33** [libktx/ktx#467] [Q&A] [answered]
In the [implementation](https://github.com/libktx/ktx/blob/2eeca22e09566ea6a101146f2634105de75c8f6f/actors/src/main/kotlin/ktx/actors/events.kt#L310) there's a listener with only `touchDown()` and `touchUp()` overridden. Is this the intended behavior? If yes, then how do I receive touchDragged events Kotlin-style? If no, then the fix is super simple

**Comment by czyzby** [comment, marked as answer]
Judging by the documentation and other available methods, this is the expected behavior. The method is not supposed to handle all touch events, just the start and end ones. As for examples of handling events in Kotlinesque style, I think the whole file you linked serves as a good example of how you can approach this. I haven't verified it in IDE, but an event handler utility specific to dragging events would look somewhat like this:

```kotlin
inline fun <T : Actor> T.onTouchDragged(crossinline listener: T.() -> Unit): InputListener {
  val inputListener = object : KtxInputListener() {
    override fun touchDragged(event: InputEvent, x: Float, y: Float, pointer: Int, button: Int) = listener()
  }
  addListener(inputListener)
  return inputListener
}
```

See the `events.kt` file for more examples of listeners with varying numbers of parameters and handled events.
